src/istio/control/http/attributes_builder.h

```diff
@@ -27,7 +27,8 @@ namespace http {
 // The context for each HTTP request.
 class AttributesBuilder {
  public:
-  AttributesBuilder(RequestContext* request) : request_(request) {}
+  AttributesBuilder(istio::mixer::v1::Attributes* attributes)
```

Comment:
question: can we avoid a raw pointer here? what stops us from using references?

Reply:
Yes, I'd love to get rid of the raw pointers.

Initially I tried reworking all this code to use the CheckContext shared pointer, but it was a massive diff.   So I exercised restraint.  And Mixer v2 is coming.

Reply:
Ok, thanks for explaining. Whether or v2 is coming soon, this is production code. I am a bit concerned about the loss of context for the lifetime of the parameter, but if we can validate that the filter never outlives the attributes with integration tests under dynamic xDS reloads, it's fine. dynamic_listener test in istio is supposed to test for that, so if it passes, then it's probably OK.

Reply:
Completely agree on the use of raw pointers.  I would love to do this, but I want to minimize this overly large PR and I think the current PR makes things no worse.  I have also verified that none of these raw pointers are used beyond the lifetime of the calling function.

So I'll do it if you guys are willing to review all the changes.   I think a separate PR on top of this would make that easier.   @mandarjog since he's also reviewing this.

Reply:
@PiotrSikora came up with a similar criticism when we were discussing a breakout of this PR.  See https://github.com/istio/proxy/pull/2117

I will do this (and I really would love to!) if you agree to review the resulting changes.  I think it would be good to do this in a new PR after 2117 is merged or else it'll become even larger